Sources/Core/ObjectiveCDebugExtension.swift

```diff
@@ -10,20 +10,42 @@ import Foundation
 
 extension ObjCModelRenderer {
     func renderDebugDescription() -> ObjCIR.Method {
-        let props = properties.map { (param, prop) -> String in
+        let props = properties.filter { (_, schema) -> Bool in
+            switch schema.schema {
+            case .boolean:
+                return false
+            default:
+                return true
+            }
+        }.map { (param, prop) -> String in
             ObjCIR.ifStmt("props.\(dirtyPropertyOption(propertyName: param, className: self.className))") {
                 let ivarName = "_\(Languages.objectiveC.snakeCaseToPropertyName(param))"
                 return ["[descriptionFields addObject:[\((ivarName + " = ").objcLiteral()) stringByAppendingFormat:\("%@".objcLiteral()), \(renderDebugStatement(param, prop.schema))]];"]
             }
         }.joined(separator: "\n")
 
+        let boolProps = properties.filter { (_, schema) -> Bool in
+            switch schema.schema {
+            case .boolean:
+                return true
+            default:
+                return false
+            }
+        }.map { (param, _) -> String in
+            ObjCIR.ifStmt("props.\(dirtyPropertyOption(propertyName: param, className: self.className))") {
+                let ivarName = "_\(booleanPropertiesIVarName).\(booleanPropertyOption(propertyName: param, className: self.className))"
+                return ["[descriptionFields addObject:[@\"\(ivarName) = \" stringByAppendingFormat:\("%@".objcLiteral()), @(\(ivarName))]];"]
+            }
+        }.joined(separator: "\n")
+
         let printFormat = "\(className) = {\\n%@\\n}".objcLiteral()
         return ObjCIR.method("- (NSString *)debugDescription") { [
             "NSArray<NSString *> *parentDebugDescription = [[super debugDescription] componentsSeparatedByString:\("\\n".objcLiteral())];",
             "NSMutableArray *descriptionFields = [NSMutableArray arrayWithCapacity:\(self.properties.count)];",
             "[descriptionFields addObject:parentDebugDescription];",
             !self.properties.isEmpty ? "struct \(self.dirtyPropertyOptionName) props = _\(self.dirtyPropertiesIVarName);" : "",
             props,
+            boolProps,
```

Comment:
If `boolProps` is empty here, the render may create those unnecessary newlines you see getting added above

Reply:
+1

Reply:
Agreed; I will fix this on my next round.